### ISSUE TITLE
appender-tracing: configurable collision policy for span-attribute enrichment

### ISSUE THREAD

**Issue by cijothomas** [open-telemetry/opentelemetry-rust#3498]
## Context

PR #3482 stabilized tracing-span attribute enrichment in `opentelemetry-appender-tracing`. With enrichment enabled, attributes from active `tracing::span!`s are copied onto each emitted log record alongside the event's own fields.

This raises the probability of duplicate attribute keys in a single log record:

- A key set on a `tracing::span!` can collide with a field on the inner `tracing::event!`.
- The same key on a parent and a child span will both be copied.

Today the appender appends in deterministic order (root → leaf → event) and relies on the SDK to handle duplicates. The SDK does not do this either today (tracked in #3497).

## Proposal

Independently of the SDK-level fix, give the appender a configuration knob to control collision behavior on the enrichment path specifically — for users who want guaranteed-unique keys without paying the dedup cost on every other code path.

Strawman:

```rust
pub enum SpanAttributeCollisionPolicy {
    /// Append in root → leaf → event order; let the SDK handle dedup. (default)
    AppendInOrder,
    /// Last-write-wins: event > leaf > root.
    LastWins,
}

OpenTelemetryTracingBridge::builder(&provider)
    .with_span_attributes(SpanAttributes::all())
    .with_span_attribute_collision_policy(SpanAttributeCollisionPolicy::LastWins)
    .build();
```

Open questions:

- Whether this is needed at all once #3497 lands (it may be redundant).
- Naming and exact semantics of each variant.
- Whether to apply only to span-vs-event collisions, only to span-vs-span, or both.

## Refs

- PR #3482 — feature stabilization: https://github.com/open-telemetry/opentelemetry-rust/pull/3482
- Discussion that surfaced this: https://github.com/open-telemetry/opentelemetry-rust/pull/3482#discussion_r3197558107
- Related: #3497 (SDK-level dedup)

**Comment by scottgerring** [comment]
Agreed that we should defer this until #3497 is resolved; if we normalise a resolution at the SDK level there that should cover the common collision case - I reckon we can wait to make this fancier until someone asks for it

**Comment by scottgerring** [comment]
Hey @hemnath-hub , we don't support that bot, and this issue is otherwise blocked (see conversation above)

**Comment by hemnath-hub** [comment]
Hi @scottgerring, thanks for the update! Since the /assign bot isn't supported here, could you assign this issue to me directly when it's unblocked? Happy to wait until the blocking conversation is resolved — just let me know if there's anything I can help with in the meantime.
